sphinx/util/logging.py

```diff
@@ -583,13 +583,13 @@ def __init__(self, stream: IO[str]) -> None:
         self.stream = stream
         self.encoding = getattr(stream, 'encoding', 'ascii') or 'ascii'
 
-    def write(self, data: str) -> None:
+    def write(self, data: str) -> int:
         try:
-            self.stream.write(data)
+            return self.stream.write(data)
         except UnicodeEncodeError:
             # stream accept only str, not bytes.  So, we encode and replace
             # non-encodable characters, then decode them.
-            self.stream.write(
+            return self.stream.write(
                 data.encode(self.encoding, 'replace').decode(self.encoding)
             )
```

Comment:
@srittau what is the return value here meant to mean? In one branch it is the number of characters written, in the other it is the number of bytes. This doesn't make sense. Please can we keep `None` as a valid return value for the (small-p) protocol?

A